support-modules/acquisition-events/src/main/scala/com/gu/support/acquisitions/bigquery/AcquisitionDataRowMapper.scala

```diff
@@ -1,26 +1,30 @@
 package com.gu.support.acquisitions
 
 import com.gu.support.acquisitions.models._
-import com.gu.support.workers._
 import org.joda.time.format.ISODateTimeFormat
+import org.json.{JSONArray, JSONObject}
 import scala.jdk.CollectionConverters._
 
-import java.util
-
 object AcquisitionDataRowMapper {
+  def mapToTableRow(acquisition: AcquisitionDataRow): JSONObject = {
 
-  def mapToTableRow(acquisition: AcquisitionDataRow): util.Map[String, Any] = {
-
-    val optionalFields = List(
+    val optionalFields: Map[String, Object] = List(
       acquisition.promoCode.map("promo_code" -> _),
       acquisition.paymentProvider.map("payment_provider" -> _.value),
       acquisition.printOptions.map(p =>
-        "print_options" -> Map(
-          "product" -> p.product.value,
-          "delivery_country_code" -> p.deliveryCountry.alpha2,
-        ).asJava,
+        "print_options" -> new JSONObject(
+          Map(
+            "product" -> p.product.value,
+            "delivery_country_code" -> p.deliveryCountry.alpha2,
+          ).asJava,
+        ),
       ),
-      acquisition.amount.map("amount" -> _),
+      acquisition.amount.map({ amount: BigDecimal =>
+        {
+          val asDouble: java.lang.Double = amount.doubleValue.toDouble
```

Comment:
While writing it I had some confusing compiler errors, so I pulled it out as an explicit val. I think what I was missing at the time was the type signature: it needs to be a `Double` rather than `double`, to be a subtype of `Object`.

Revising now, it looks like I can remove the explicit val alright, giving the following:

```
      acquisition.amount.map({ amount: BigDecimal =>
        "amount" -> (amount.doubleValue.toDouble: java.lang.Double)
      }),
```

But I can’t figure out how to get it to work without the type signature, or with the underscore syntax.

Reply:
I wonder: does `asJava` do what I need for doubles?

Reply:
No, it doesn’t look like it.